src/main/java/org/mule/service/http/impl/service/server/grizzly/GrizzlyServerManager.java

```diff
@@ -196,13 +196,12 @@ public HttpServer createSslServerFor(TlsContextFactory tlsContextFactory, Suppli
   @Override
   public HttpServer createServerFor(ServerAddress serverAddress, Supplier<Scheduler> schedulerSupplier,
                                     boolean usePersistentConnections, int connectionIdleTimeout, ServerIdentifier identifier)
-      throws IOException {
+      throws ServerCreationException {
     if (logger.isDebugEnabled()) {
       logger.debug("Creating http server socket for ip {} and port {}", serverAddress.getIp(), serverAddress.getPort());
     }
     if (servers.containsKey(serverAddress)) {
-      throw new IllegalStateException(String.format("Could not create a server for %s since there's already one.",
-                                                    serverAddress));
+      throw new ServerCreationException(format("There's already a server for %s.", serverAddress));
```

Comment:
This is slightly different because the other checks for 0.0.0.0 and I think names as well. I think this is just a double check. I do think it makes sense to have a subtype of ServerCreationException for these cases, even if the message is different.